pkg/kube/client.go

```diff
@@ -319,9 +325,63 @@ func updateResource(target *resource.Info, currentObj runtime.Object) error {
 	// send patch to server
 	helper := resource.NewHelper(target.Client, target.Mapping)
 	_, err = helper.Patch(target.Namespace, target.Name, api.StrategicMergePatchType, patch)
+
+	if err != nil {
+		return err
+	}
+
+	if restart {
+		kind := target.Mapping.GroupVersionKind.Kind
+
+		client, _ := c.ClientSet()
+		switch kind {
```

Comment:
I found that implementation of this method doesn't cover cases with daemon and statefulsets. Moreover, it also returns labels for generic pods which I tried to prevent. So we again have to have something like switch for needed kinds.